node/core/approval-voting/src/lib.rs

```diff
@@ -1452,8 +1494,12 @@ async fn handle_approved_ancestor<Context>(
 				unapproved.len(),
 				entry.candidates().len(),
 			);
-
+			entry_span.add_uint_tag("unapproved-candidates", unapproved.len() as u64);
 			for candidate_hash in unapproved {
+				let mut candidate_span = entry_span
```

Comment:
This operation will be really fast and will really spam with spans. The root span `handle-approved-ancestor` should be enough.